### PR TITLE
split out the Consul TF profile

### PR DESCRIPTION
**Pull request by phrawzty**
It was a long and painful road to get to this deceptively simple-looking PR.

@rhelmer `r?`

### CODE REVIEW

File: puppet/modules/socorro/files/etc_consul.d/config.json [hunk 1]
```diff
@@ -1,5 +1,6 @@
 {
     "server": true,
+    "bootstrap_expect": 3,
```
Comment (phrawzty) [inline puppet/modules/socorro/files/etc_consul.d/config.json]: If we ever want more than 3 consul servers we'll have to adjust this.

File: puppet/modules/socorro/manifests/role/consul.pp [hunk 2]
```diff
@@ -9,8 +9,11 @@
   }
 
   package {
-    'consul':
-      ensure => latest
+    [
+      'bind-utils',
```
Comment (phrawzty) [inline puppet/modules/socorro/manifests/role/consul.pp]: `bind-utils` is not necessary but it's handy for testing Consul, so...

File: terraform/consul/main.tf [hunk 3]
```diff
@@ -0,0 +1,107 @@
+# CONSUL CLUSTER ONLY. NOTHING ELSE BELONGS HERE.
+provider "aws" {
+    region = "${var.region}"
+    access_key = "${var.access_key}"
+    secret_key = "${var.secret_key}"
+}
+
+# Source: https://consul.io/docs/agent/options.html
+resource "aws_security_group" "private_to_consul__consul" {
```
Comment (phrawzty) [inline terraform/consul/main.tf]: I figure that someday we'll tighten up the internal ACLs (i.e. remove `aws_security_grpup.private_to_private__any`), so this is actually a step in the *right* direction.

At least, that's what I have to tell myself. :persevere:

File: terraform/consul/main.tf [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+        cidr_blocks = [
+            "172.31.0.0/16"
+        ]
+    }
+    tags {
+        Environment = "${var.environment}"
+    }
+}
+
+resource "aws_security_group" "internet_to_consul__ssh" {
```
Comment (phrawzty) [inline terraform/consul/main.tf]: This ends up being a duplicate of `aws_security_group.internet_to_any`. :unamused:

File: terraform/consul/main.tf (hunk 3 above)
Comment (rhelmer) [inline terraform/consul/main.tf]: +1, yeah we could be a lot more explicit about what can connect where.

File: terraform/main.tf [hunk 5]
```diff
@@ -14,24 +14,24 @@ resource "aws_security_group" "private_to_private__any" {
         protocol = "tcp"
         cidr_blocks = [
             "172.31.0.0/16"
-            ]
-        }
+        ]
```
Comment (phrawzty) [inline terraform/main.tf]: The spacing on these bothered me, so I fixed them. :facepunch:

**Comment by rhelmer** [comment]
@phrawzty we should have the travis-ci job do a ```terraform plan``` in the consul dir too

**Comment by rhelmer** [comment]
r+

**Pull request merged**